content/guides/caching.md

```diff
@@ -4,284 +4,356 @@ sort: 11
 
-## Get filenames from webpack compilation stats
-
-When working in development mode, you just reference JavaScript files by entry point name in your HTML.
-
-```html
-<script src="vendor.js"></script>
-<script src="main.js"></script>
+  module.exports = {
+    entry: './src/index.js',
+    plugins: [
+      new CleanWebpackPlugin(['dist']),
+      new HtmlWebpackPlugin({
+        title: 'Caching'
+-     })
++     }),
++     new webpack.optimize.CommonsChunkPlugin({
++       name: 'runtime'
++     })
+    ],
+    output: {
+      filename: '[name].[contenthash].js',
+      path: path.resolve(__dirname, 'dist')
+    }
+  };
 ```
 
-Although, each time we build for production, we’ll get different file names. Something, that looks like this:
+Let's run another build to see the extracted `runtime` bundle:
 
-```html
-<script src="vendor.50cfb8f89ce2262e5325.js"></script>
-<script src="main.70b594fe8b07bcedaa98.js"></script>
+``` bash
+Hash: 80552632979856ddab34
+Version: webpack 3.3.0
+Time: 1512ms
+                          Asset       Size  Chunks                    Chunk Names
+   main.5ec8e954e32d66dee1aa.js     542 kB       0  [emitted]  [big]  main
+runtime.719796322be98041fff2.js    5.82 kB       1  [emitted]         runtime
+                     index.html  275 bytes          [emitted]
+   [0] ./src/index.js 336 bytes {0} [built]
+   [2] (webpack)/buildin/global.js 509 bytes {0} [built]
+   [3] (webpack)/buildin/module.js 517 bytes {0} [built]
+    + 1 hidden module
 ```
 
-In order to reference a correct file in the HTML, we’ll need information about our build. This can be extracted from webpack compilation stats by using this plugin:
-
-```js
-// webpack.config.js
-const path = require("path");
-
-module.exports = {
-  /*...*/
-  plugins: [
-    function() {
-      this.plugin("done", function(stats) {
-        require("fs").writeFileSync(
-          path.join(__dirname, "build", "stats.json"),
-          JSON.stringify(stats.toJson()));
-      });
+It's also good practice to extract third-party libraries, such as `lodash` or `react`, to a separate `vendor` chunk as they are less likely to change than our local source code. This step will allow clients to request even less from the server to stay up to date. This can be done by using a combination of a new `entry` point along with another `CommonsChunkPlugin` instance:
+
+__webpack.config.js__
+
+``` diff
+  var path = require('path');
+  const webpack = require('webpack');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+  module.exports = {
+-   entry: './src/index.js',
++   entry: {
++     main: './src/index.js',
++     vendor: [
++       'lodash'
++     ]
++   },
+    plugins: [
+      new CleanWebpackPlugin(['dist']),
+      new HtmlWebpackPlugin({
+        title: 'Caching'
+      }),
++     new webpack.optimize.CommonsChunkPlugin({
++       name: 'vendor'
```

Comment:
See the example at TheDutchCoder/webpack-guides-code-examples#17... I don't think it's necessary but I also didn't test chunks too much yet (though I do have a bit more code to push that uses `import()` to generate a child chunk).